### ISSUE TITLE
Disable inotify when audit based events are enabled.

### ISSUE THREAD

**Issue by kpm4** [osquery/osquery#5584]
# Feature request

### What new feature do you want?

hello, I noticed that there is no way to enable FIM with auditd only, so it would be nice to have a flag to disable inotify.
disable_events=false  will enable inotify and auditd based events.

### How is this new feature useful?

Having inotify and auditd based event for FIM can degrade the performances. The ability to choose which one enable seems useful.

### How can this be implemented?

something like disable_inotify=true

**Comment by mike-myers-tob** [comment]
When running with `--disabled_events=false` in osquery 4.5.1, it looks like the Audit publisher defaults to disabled (and can be enabled with `--disable_audit=false`), but the inotify and udev publishers both default to enabled.

```
$ sudo osqueryi --disable_events=false
Using a virtual database. Need help, type '.help'
osquery> select * from osquery_events;
+---------------------+---------------------+------------+---------------+--------+-----------+--------+
| name                | publisher           | type       | subscriptions | events | refreshes | active |
+---------------------+---------------------+------------+---------------+--------+-----------+--------+
| auditeventpublisher | auditeventpublisher | publisher  | 2             | 0      | 0         | 0      |
| inotify             | inotify             | publisher  | 0             | 0      | 2         | 1      |
| syslog              | syslog              | publisher  | 1             | 0      | 0         | 0      |
| udev                | udev                | publisher  | 1             | 0      | 2         | 1      |
| apparmor_events     | auditeventpublisher | subscriber | 0             | 0      | 0         | 0      |
| file_events         | inotify             | subscriber | 0             | 0      | 0         | 1      |
| hardware_events     | udev                | subscriber | 1             | 0      | 0         | 1      |
| process_events      | auditeventpublisher | subscriber | 1             | 0      | 0         | 1      |
| process_file_events | auditeventpublisher | subscriber | 0             | 0      | 0         | 0      |
| selinux_events      | auditeventpublisher | subscriber | 0             | 0      | 0         | 0      |
| socket_events       | auditeventpublisher | subscriber | 0             | 0      | 0         | 0      |
| syslog_events       | syslog              | subscriber | 1             | 0      | 0         | 1      |
| user_events         | auditeventpublisher | subscriber | 1             | 0      | 0         | 1      |
| yara_events         | inotify             | subscriber | 0             | 0      | 0         | 1      |
+---------------------+---------------------+------------+---------------+--------+-----------+--------+
```

**Comment by mike-myers-tob** [comment]
Fixed with #6663
